tasmota/tasmota_xsns_sensor/xsns_109_sgp41.ino

```diff
@@ -0,0 +1,201 @@
+/*
+  xsns_109_sgp41.ino - SGP41 VOC and NOx sensor support for Tasmota
+
+  Copyright (C) 2023  Andrew Klaus
+
+  This program is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+
+  This program is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#ifdef USE_I2C
+#ifdef USE_SGP41
+/*********************************************************************************************\
+ * SGP41 - Gas (TVOC - Total Volatile Organic Compounds) and NOx
+ *
+ * Source: Gerhard Mutz and  Industries
+ * Adaption for TASMOTA: Andrew Klaus
+ *
+ * I2C Address: 0x59
+\*********************************************************************************************/
+
+#define XSNS_109            109
+#define XI2C_82             82  // See I2CDEVICES.md
+
+#define SGP41_ADDRESS       0x59
+
+#include "SensirionI2CSgp41.h"
+
+extern "C" {
+#include "sensirion_gas_index_algorithm.h"
+};
+
+SensirionI2CSgp41 sgp41;
+
+bool sgp41_init = false;
+bool sgp41_ready = false;
+uint16_t srawVoc;
+uint16_t srawNox;
+int32_t voc_index_sgp41;
+int32_t nox_index_sgp41;
+
+uint16_t conditioning_s = 1;
+
+GasIndexAlgorithmParams voc_algorithm_params;
+GasIndexAlgorithmParams nox_algorithm_params;
+
+/********************************************************************************************/
+void sgp41_print_serial(uint16_t serialNumber[], uint8_t serialNumberSize) {
```

Comment:
Please remove any `Serial.print()`. We must not pollute the serial bus with such logs.

Use `AddLog()` to emit proper logs.